x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/utils/get_filter.ts

```diff
@@ -59,7 +61,11 @@ export const getFilter = async ({
   query,
   exceptionFilter,
   fields = [],
```

Comment:
A note for follow up from security: we should split this function into 3 functions, one that loads fields automatically, one that requires fields to be passed in, and one that does not handle fields at all. Then we should explicitly call the version we want for each use case instead of trying to ensure that the parameters are passed in as expected.

Reply:
Hey @marshallmain - can you open a ticket to track this a tech debt?

Reply:
https://github.com/elastic/kibana/issues/185890